pkgs/servers/x11/xorg/overrides.nix

```diff
@@ -763,8 +763,15 @@ self: super:
             sha256 = "0zm9g0g1jvy79sgkvy0rjm6ywrdba2xjd1nsnjbxjccckbr6i396";
             name = "revert-fb-changes-2.patch";
           })
+          ./darwin/bundle_main.patch
+          ./darwin/stub.patch
         ];
 
+        postPatch = attrs.postPatch + ''
+          substituteInPlace hw/xquartz/mach-startup/stub.c \
+            --subst-var-by XQUARTZ_APP "$out/Applications/XQuartz.app"
```

Comment:
I guess it's nicer to do this in `patches` using `substituteAll`.

Reply:
I don't understand what you mean. I can't directly embed a `$out` path in a patch file.

Reply:
Instead of doing

```nix
patches = [
  ./darwin/stub.patch
];
```
You can put something like

```nix
patches = [
  (substituteAll {
    src = ./darwin/stub.patch;
    XQUARTZ_APP = "${placeholder "out"}/Applications/XQuartz.app";
  })
];
```

Reply:
I'm not sure why but I couldn't get `subsituteAll` to work. I pushed a change using `substitute` instead.